website/project/model.py

```diff
@@ -1239,18 +1239,20 @@ def set_node_license(self, license_id, year, copyright_holders, auth, save=True)
         record.copyright_holders = copyright_holders or []
         record.save()
         self.node_license = record
-        self.add_log(
-            action=NodeLog.CHANGED_LICENSE,
-            params={
-                'parent_node': self.parent_id,
-                'node': self._primary_key,
-                'new_license': node_license.name
-            },
-            auth=auth,
-            save=False,
-        )
-        if save:
-            self.save()
+
+        for node in self.node_and_primary_descendants():
+            node.add_log(
+                action=NodeLog.CHANGED_LICENSE,
+                params={
+                    'parent_node': node.parent_id,
+                    'node': node._primary_key,
+                    'new_license': node_license.name
+                },
+                auth=auth,
+                save=False,
+            )
+            if save:
```

Comment:
I'm not sure this check is desired. `save=False` is useful for preventing a save of the primary object. Any secondary objects (e.g. descendant nodes) should be saved automatically.

This is to prevent a situation where not all nodes will get saved.

```python
node.set_node_license(... save=False)
# ...
node.save()
```

In this case, the descendant nodes would not be saved.

One workaround would be to skip saving `self` in the loop:

```python
for node in self.node_and_primary_descendants():
    # add log....
    if node._id != self._id:
        node.save()

if save:
    self.save()
```